compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2404,21 +2476,21 @@ object Parsers {
      *                  | InfixExpr id ColonArgument
      *                  | InfixExpr MatchClause
      */
-    def postfixExpr(location: Location = Location.ElseWhere): Tree =
-      val t = postfixExprRest(prefixExpr(location), location)
+    def postfixExpr(location: Location = Location.ElseWhere, inStatSeq: Boolean = false): Tree =
+      val t = postfixExprRest(prefixExpr(location, inStatSeq), location)
       if location.inArgs && followingIsVararg() then
-        Typed(t, atSpan(in.skipToken()) { Ident(tpnme.WILDCARD_STAR) })
+        Typed(t, atSpan(skipToken()) { Ident(tpnme.WILDCARD_STAR) })
       else
         t
 
     def postfixExprRest(t: Tree, location: Location): Tree =
-      infixOps(t, in.canStartExprTokens, prefixExpr, location, ParseKind.Expr,
+      infixOps(t, in.canStartExprTokens, prefixExpr(_), location, ParseKind.Expr,
         isOperator = !(location.inArgs && followingIsVararg()))
 
     /** PrefixExpr       ::= [PrefixOperator'] SimpleExpr
      *  PrefixOperator   ::=  ‘-’ | ‘+’ | ‘~’ | ‘!’ (if not backquoted)
      */
-    val prefixExpr: Location => Tree = location =>
```

Comment:
The `val` is there for performance. One closure per parser instead of one for every expression in the program. We should keep it that way,